tools/topology/platform/common/ssp.m4

```diff
@@ -30,6 +30,11 @@ $6
 dnl SSP_QUIRK_LBM 64 = (1 << 6)
 define(`SSP_QUIRK_LBM', 64)
 
+dnl SSP_CC_MCLK_ES 64 = (1 << 6)
+define(`SSP_CC_MCLK_ES', 64)
+dnl SSP_CC_BCLK_ES 128 = (1 << 7)
+define(`SSP_CC_BCLK_ES', 128)
+
 dnl SSP_CONFIG_DATA(type, idx, valid bits, mclk_id, quirks, bclk_delay,
 dnl clks_control, pulse_width, padding)
 dnl mclk_id, quirks, bclk_delay clks_control, pulse_width and padding are optional
```

Comment:
it's incomplete, please check
https://github.com/thesofproject/sof/pull/4249/commits/8e67cb3e073eb353096adceea5dc65c247f40acd

Reply:
i think this patchset isn't working since the topology is incorrect...

Reply:
> it's incomplete, please check
8e67cb3

@brentlu Is https://github.com/thesofproject/sof/commit/49547404e98a867576b7ba147ff902604f5c4f2d the missing part?

Reply:
i think that patch is meant to fix empty string like this, it should be ok not having the https://github.com/thesofproject/sof/commit/49547404e98a867576b7ba147ff902604f5c4f2d

SectionVendorTuples."DAICONFIG.SSP1_tuples" {
	tokens "sof_ssp_tokens"
	tuples."word" {
		SOF_TKN_INTEL_SSP_SAMPLE_BITS	"16"
		SOF_TKN_INTEL_SSP_QUIRKS	""
		SOF_TKN_INTEL_SSP_CLKS_CONTROL	""
	}
	tuples."short" {
		SOF_TKN_INTEL_SSP_MCLK_ID	"1"
	}
}

Reply:
> i think that patch is meant to fix empty string like this, it should be ok not having the [4954740](https://github.com/thesofproject/sof/commit/49547404e98a867576b7ba147ff902604f5c4f2d)
> 
> SectionVendorTuples."DAICONFIG.SSP1_tuples" {
> tokens "sof_ssp_tokens"
> tuples."word" {
> SOF_TKN_INTEL_SSP_SAMPLE_BITS "16"
> SOF_TKN_INTEL_SSP_QUIRKS ""
> SOF_TKN_INTEL_SSP_CLKS_CONTROL ""
> }
> tuples."short" {
> SOF_TKN_INTEL_SSP_MCLK_ID "1"
> }
> }

Sorry @brentlu I didn't get your point. Could you share a patch to describe what is incorrect or missing?